allennlp/training/trainer.py

```diff
@@ -156,7 +156,9 @@ def str_to_time(time_str: str) -> datetime.datetime:
     return datetime.datetime(*pieces)
 
 
-class Trainer:
+class Trainer(Registrable):
```

Comment:
You need to add `# type: ignore` and `# pylint: disable=arguments-differ` on `from_params` in this class.  Pylint and mypy are failing because of this.